test/files/neg/userdefined_apply.scala

```diff
@@ -0,0 +1,31 @@
+object ClashOverloadNoSig {
+  // error: overloaded method apply needs result type
+  private def apply(x: Int) = if (x > 0) new ClashOverloadNoSig(x) else apply("")
+
+  def apply(x: String): ClashOverloadNoSig = ???
+}
+
+case class ClashOverloadNoSig private(x: Int)
+
+object ClashRecNoSig {
+  // error: recursive method apply needs result type
```

Comment:
The actual report says `overloaded method apply needs result type`, as you explain in the lengthy comment in Namers. So the behavior is fine, just the comment here is out of synch.